market/connector/docusign/demo/eSignatureDemo/src/net/docusign/auth/OAuth2Feature.java

```diff
@@ -48,37 +45,37 @@ public boolean configure(FeatureContext context)
   {
     var config = new FeatureConfig(context.getConfiguration(), OAuth2Feature.class);
     var docuSignUri = new OAuth2UriProperty(config, Property.AUTH_BASE_URI, "https://account-d.docusign.com/oauth");
-    ISession session = Ivy.session();
     var oauth2 = new OAuth2BearerFilter(
-      ctxt -> requestToken(ctxt, docuSignUri, session), 
+      ctxt -> requestToken(ctxt, docuSignUri), 
       docuSignUri
     );
     context.register(oauth2, Priorities.AUTHORIZATION);
-    context.register(new UserUriFilter(session, docuSignUri), Priorities.AUTHORIZATION+10);
+    context.register(new UserUriFilter(ISession.current(), docuSignUri), Priorities.AUTHORIZATION+10);
     return true;
   }
   
-  private static Response requestToken(AuthContext ctxt, OAuth2UriProperty uriFactory, ISession session)
+  private static Response requestToken(AuthContext ctxt, OAuth2UriProperty uriFactory)
   {
-    if (session.getIdentifier() == ISecurityConstants.SYSTEM_USER_SESSION_ID)
+    ISession current = ISession.current();
+    if (current.getIdentifier() == ISecurityConstants.SYSTEM_USER_SESSION_ID)
     {
-      return jwtGrantToken(ctxt);
+      return jwtGrantToken(ctxt, uriFactory);
     }
     return webUserGrantToken(ctxt, uriFactory);
   }
 
   private static Response webUserGrantToken(AuthContext ctxt, OAuth2UriProperty uriFactory)
   {
-    String authCode = IRequest.current().getFirstParameter("code");
-    if (StringUtils.isBlank(authCode))
+    var authCode = ctxt.authCode();
+    if (authCode.isEmpty())
```

Comment:
is authCode never null? If it can be null StringUtils.isBlank is better I think....